ivy/stateful/module.py

```diff
@@ -428,6 +447,76 @@ def found_dup_callback(x, kc):
             vs = vs.cont_prune_key_chain(dup_kc)
         return vs, keychain_mappings
 
+    def _buffer_tracker(self, func):
+        """Tracks the variables defined as buffer variables and stores them."""
+
+        def wrapper(*args, **kwargs):
+            initial_object_snapshot = self.__dict__.copy()
+            # initialise the buffers
+            func(*args, **kwargs)
+            final_object_snapshot = self.__dict__
+            getattr(self, "buffers", {}).update(
+                set(final_object_snapshot.keys()).difference(
+                    set(initial_object_snapshot.keys())
+                )
+            )
+
+        wrapper.buffers_tracked = True
+        return wrapper
+
+    def _get_buffers(self):
+        """Return the buffer variables, if any, of the given Module class instance."""
+        if self.buffers:
+            buffer_dict = {}
+            for buffer in self.buffers:
+                buffer_dict[buffer] = (
+                    getattr(self, buffer)
+                    if not isinstance(getattr(self, buffer), ivy.Module)
+                    else getattr(self, buffer)._get_buffers()
+                )
+            return buffer_dict
+        return {}
+
+    def _set_buffers(self, buffers, override=False, nesting=None):
+        """
+        Set the buffers of the given class instance, according to the buffers passed.
+
+        Parameters
+        ----------
+        buffers
+            a dictionary with variable names and corresponding values
+
+        override
+            if true, sets the variable as an attribute even if it doesn't exist
+        """
+        for buffer in buffers:
+            if hasattr(self, buffer):
+                # check if this value is another nested dictionary, if yes
+                # we recurse
+                if isinstance(buffers[buffer], dict):
+                    getattr(self, buffer)._set_buffers(
+                        buffers=buffers[buffer], override=override
+                    )
+                else:
+                    setattr(self, buffer, buffers[buffer])
+            elif not override or (override and isinstance(buffers[buffer], dict)):
+                # do a quick check to see if this is nested case
+                # we can't set buffers for module objects not yet
+                # created, even if override=True
+                raise ivy.exceptions.IvyNotImplementedException(
+                    f"{buffer} hasn't been defined for the given Module structure"
+                )
+            else:
+                setattr(self, buffer, buffers[buffer])
+                if hasattr(self, "buffers"):
+                    self.buffers.update({buffer})
+                else:
+                    setattr(self, "buffers", {buffer})
+
+    def _register_buffers(self, var_name, value):
```

Comment:
make this public, and remove variables.

Reply:
could you please rename `register_buffers` to `register_buffer`, `set_buffers` to `set_buffer` and others and make the necessary ones public?